src/parser.ts

```diff
@@ -3130,18 +3132,15 @@ class Parser {
                 } else {
                     hasConstructor.value = true;
                 }
-                property.kind = 'constructor';
+                kind = 'constructor';
             }
         } else {
-            if (!property.computed && (property.key.name || property.key.value.toString()) === 'prototype') {
+            if (!computed && this.isPropertyKey(key, 'prototype')) {
```

Comment:
Good point. And now that I've taken a second look, I think the conditional block needs be based on `!computed` anyway for improved readability.